hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/ReconRocksDB.java

```diff
@@ -17,79 +17,121 @@
+            reconUtils.getLastKnownDB(reconDbDir, RECON_CONTAINER_KEY_DB);
     if (lastKnownContainerKeyDb != null) {
-      LOG.info("Last known container-key DB : {}",
-          lastKnownContainerKeyDb.getAbsolutePath());
-      dbStore = initializeDBStore(configuration,
-          lastKnownContainerKeyDb.getName());
+      LOG.info("Last known DB : {}",
+              lastKnownContainerKeyDb.getAbsolutePath());
+      db = initializeDBStore(configuration,
+              lastKnownContainerKeyDb.getName());
     } else {
-      dbStore = getNewDBStore(configuration);
+      db = getNewDBStore(configuration);
     }
-    if (dbStore == null) {
+    if (db == null) {
       throw new ProvisionException("Unable to provide instance of DBStore " +
-          "store.");
+              "store.");
     }
+    return db;
+  }
+
+  public DBStore getDbStore() {
     return dbStore;
   }
 
+  public void reInit() throws IOException {
+    File oldDBLocation = dbStore.getDbLocation();
+    try {
+      dbStore.close();
+    } catch (Exception e) {
+      LOG.warn("Unable to close old Recon container key DB at {}.",
+              dbStore.getDbLocation().getAbsolutePath());
+    }
+    dbStore = getNewDBStore(configuration);
+    LOG.info("Creating new Recon Container DB at {}",
+            dbStore.getDbLocation().getAbsolutePath());
+
+    if (oldDBLocation.exists()) {
+      LOG.info("Cleaning up old Recon Container key DB at {}.",
+              oldDBLocation.getAbsolutePath());
+      FileUtils.deleteDirectory(oldDBLocation);
+    }
+  }
+
+  static void clearTable(Table table) throws IOException {
+    if (table == null) {
+      return;
+    }
+    TableIterator<Object, ? extends KeyValue<Object, Object>>
+            tableIterator = table.iterator();
+    while (tableIterator.hasNext()) {
+      KeyValue<Object, Object> entry = tableIterator.next();
+      table.delete(entry.getKey());
+    }
+  }
+
+  static DBStore getNewDBStore(OzoneConfiguration configuration) {
+    String dbName = RECON_CONTAINER_KEY_DB + "_" + System.currentTimeMillis();
+    return initializeDBStore(configuration, dbName);
+  }
+
   private static DBStore initializeDBStore(OzoneConfiguration configuration,
                                            String dbName) {
     DBStore dbStore = null;
     try {
       dbStore = DBStoreBuilder.createDBStore(configuration,
-          new ReconDBDefinition(dbName));
+              new ReconDBDefinition(dbName));
     } catch (Exception ex) {
       LOG.error("Unable to initialize Recon container metadata store.", ex);
     }
     return dbStore;
   }
 
-  static DBStore getNewDBStore(OzoneConfiguration configuration) {
-    String dbName = RECON_CONTAINER_KEY_DB + "_" + System.currentTimeMillis();
-    return initializeDBStore(configuration, dbName);
+  public void close() throws Exception {
+    if (this.dbStore != null) {
+      dbStore.close();
+      dbStore = null;
+    }
   }
-}
+}
```

Comment:
nit: a new line at the end of file
```suggestion
}

```